### ISSUE TITLE
Add mobs

### ISSUE THREAD

**Issue by 22i** [XenonCoder/TerraCraft#40]
Mobs would be epic!

**Comment by XenonCoder** [comment]
@22i mobs are not planned at the moment. We want to complete the engine. Thanks to Nuitka we can release independent builds (executable binaries). This will increase performance and eventually the mobs will be inserted. Thank you.

**Comment by benmoran56** [comment]
For things like Mobs, we will need to do a few architectural changes and improvements first. It's not hard, but as Xenon said, it's not a current priority. 

Some things we need to do before adding mobs:

1.  Make a generic character class, to track positions and other attributes for living and moving things. The player could also subclass from this.

2. Add a system for AI. To keep it easy to modify, maybe this can be a module with various AI classes.

3. We should have a simple scripting system. This will be useful for other things also, such as trigger events. It can be done by making a simple Script class that has certain structure. All scripts can be subclasses of that.

**Comment by trevortomesh** [comment]
I'd be happy to work on simple mobs -- I am actually hoping to use this to allow students to write their own AI, etc.

**Comment by benmoran56** [comment]
That would be great. Before going forward, I would like to make some architectural changes, and implement a Scripting system. I have most of this code written already, so I will try to make the changes soon. 

In the mean time, we will need to create the 3D models for the mobs. In the recent version of pyglet, loading models in `obj` files is supported. That might be useful here. 
Or, perhaps something simpler to start with? For example, the mods could just be simple movable rectangles, with different textures?

**Comment by trevortomesh** [comment]
Is the scripting system in python? Please say it is!

**Comment by benmoran56** [comment]
Yes, it is. Since it's Python, it's quite trivial to implement. This is basically all there is to it: 

```
class Script:
    finished = False

    def update(self, scene):
        raise NotImplementedError


class ExampleScript(Script):

    def update(self, scene):
        print("Hello World")
        self.finished = True
```

**Comment by trevortomesh** [comment]
That'll be awesome. I am really looking forward to using this with my students in upcoming semesters!

**Comment by trevortomesh** [comment]
I think we should just start with a cube that can be scripted to move around the world. I tried doing something with a terrain block but I failed hard.

**Comment by benmoran56** [comment]
I'll try to finish the architectural changes I planned first of all. After that, it should be more easy to extend things.
